python/smqtk/algorithms/nn_index/mrpt.py

```diff
@@ -0,0 +1,451 @@
+        higher values, but at the cost of using more memory on any particular
+        query. As a rule of thumb, for a given value of `k`, num_trees should
+        be about `3k/L`.
+    """
+
+    @classmethod
+    def is_usable(cls):
+        return True
+
+    def __init__(self, index_filepath=None, parameters_filepath=None,
+                 descriptor_cache_filepath=None,
+                 # Parameters for building an index
+                 num_trees=10, depth=1, required_votes=3, random_seed=None,
+                 pickle_protocol=pickle.HIGHEST_PROTOCOL):
+        """
+        Initialize MRPT index properties. Does not contain a queryable index
+        until one is built via the ``build_index`` method, or loaded from
+        existing model files.
+
+        :param pickle_protocol: The protocol version to be used by the pickle
+            module to serialize class information
+        :type pickle_protocol: int
+
+        :param index_filepath: Optional file location to load/store MRPT index
+            when initialized and/or built.
+
+            If not configured, no model files are written to or loaded from
+            disk.
+        :type index_filepath: None | str
+
+        :param parameters_filepath: Optional file location to load/save index
+            parameters determined at build time.
+
+            If not configured, no model files are written to or loaded from
+            disk.
+        :type parameters_filepath: None | str
+
+        :param descriptor_cache_filepath: Optional file location to load/store
+            DescriptorElements in this index.
+
+            If not configured, no model files are written to or loaded from
+            disk.
+        :type descriptor_cache_filepath: None | str
+
+        :param num_trees: The number of trees that will be generated for the
+            data structure
+        :type num_trees: None | int
+
+        :param depth: The depth of the trees
+        :type depth: None | int
+
+        :param required_votes: The number of votes required for a nearest
+            neighbor
+        :type required_votes: None | int
+
+        :param random_seed: Integer to use as the random number generator seed.
+        :type random_seed: int
+
+        """
+        super(MRPTNearestNeighborsIndex, self).__init__()
+
+        self._pickle_protocol = pickle_protocol
+
+        def normpath(p):
+            return (p and osp.abspath(osp.expanduser(p))) or p
+
+        self._index_filepath = normpath(index_filepath)
+        self._index_param_filepath = normpath(parameters_filepath)
+        self._descr_cache_filepath = normpath(descriptor_cache_filepath)
+        # Now they're either None or an absolute path
+
+        # parameters for building an index
+        self._depth = depth
+        self._num_trees = num_trees
+        self._required_votes = required_votes
+
+        self._trees = []
+
+        # In-order cache of descriptors we're indexing over.
```

Comment:
No, it's so that it has a sane value in case it gets referenced. I'll add a comment.